2025-01-17 16:28:23,497 INFO    MainThread:12039 [wandb_setup.py:_flush():76] Current SDK version is 0.16.4
2025-01-17 16:28:23,497 INFO    MainThread:12039 [wandb_setup.py:_flush():76] Configure stats pid to 12039
2025-01-17 16:28:23,498 INFO    MainThread:12039 [wandb_setup.py:_flush():76] Loading settings from /Users/zhijuanzhang/.config/wandb/settings
2025-01-17 16:28:23,498 INFO    MainThread:12039 [wandb_setup.py:_flush():76] Loading settings from /Users/zhijuanzhang/Documents/UdacityCourse/MLOps_P2_Reproducible_Pipeline/Exercise/lesson-3-data-validation/exercises/exercise_9/starter/wandb/settings
2025-01-17 16:28:23,498 INFO    MainThread:12039 [wandb_setup.py:_flush():76] Loading settings from environment variables: {}
2025-01-17 16:28:23,498 INFO    MainThread:12039 [wandb_setup.py:_flush():76] Applying setup settings: {'_disable_service': False}
2025-01-17 16:28:23,498 WARNING MainThread:12039 [wandb_setup.py:_flush():76] Could not save program above cwd: /opt/anaconda3/envs/mlflow-60000e2b97642ccc38c7a9ee503977f0abd54819/bin/pytest
2025-01-17 16:28:23,498 INFO    MainThread:12039 [wandb_setup.py:_flush():76] Inferring run settings from compute environment: {'program_relpath': None, 'program_abspath': '/opt/anaconda3/envs/mlflow-60000e2b97642ccc38c7a9ee503977f0abd54819/bin/pytest', 'program': '/opt/anaconda3/envs/mlflow-60000e2b97642ccc38c7a9ee503977f0abd54819/bin/pytest'}
2025-01-17 16:28:23,498 INFO    MainThread:12039 [wandb_init.py:_log_setup():526] Logging user logs to /Users/zhijuanzhang/Documents/UdacityCourse/MLOps_P2_Reproducible_Pipeline/Exercise/lesson-3-data-validation/exercises/exercise_9/starter/wandb/run-20250117_162823-yl56zo0v/logs/debug.log
2025-01-17 16:28:23,498 INFO    MainThread:12039 [wandb_init.py:_log_setup():527] Logging internal logs to /Users/zhijuanzhang/Documents/UdacityCourse/MLOps_P2_Reproducible_Pipeline/Exercise/lesson-3-data-validation/exercises/exercise_9/starter/wandb/run-20250117_162823-yl56zo0v/logs/debug-internal.log
2025-01-17 16:28:23,498 INFO    MainThread:12039 [wandb_init.py:init():566] calling init triggers
2025-01-17 16:28:23,498 INFO    MainThread:12039 [wandb_init.py:init():573] wandb.init called with sweep_config: {}
config: {}
2025-01-17 16:28:23,498 INFO    MainThread:12039 [wandb_init.py:init():616] starting backend
2025-01-17 16:28:23,498 INFO    MainThread:12039 [wandb_init.py:init():620] setting up manager
2025-01-17 16:28:23,500 INFO    MainThread:12039 [backend.py:_multiprocessing_setup():105] multiprocessing start_methods=spawn,fork,forkserver, using: spawn
2025-01-17 16:28:23,501 INFO    MainThread:12039 [wandb_init.py:init():628] backend started and connected
2025-01-17 16:28:23,506 INFO    MainThread:12039 [wandb_init.py:init():720] updated telemetry
2025-01-17 16:28:23,507 INFO    MainThread:12039 [wandb_init.py:init():753] communicating run to backend with 90.0 second timeout
2025-01-17 16:28:23,739 INFO    MainThread:12039 [wandb_run.py:_on_init():2262] communicating current version
2025-01-17 16:28:23,804 INFO    MainThread:12039 [wandb_run.py:_on_init():2271] got version response upgrade_message: "wandb version 0.19.4 is available!  To upgrade, please run:\n $ pip install wandb --upgrade"

2025-01-17 16:28:23,804 INFO    MainThread:12039 [wandb_init.py:init():804] starting run threads in backend
2025-01-17 16:28:24,899 INFO    MainThread:12039 [wandb_run.py:_console_start():2241] atexit reg
2025-01-17 16:28:24,899 INFO    MainThread:12039 [wandb_run.py:_redirect():2096] redirect: wrap_raw
2025-01-17 16:28:24,899 INFO    MainThread:12039 [wandb_run.py:_redirect():2161] Wrapping output streams.
2025-01-17 16:28:24,899 INFO    MainThread:12039 [wandb_run.py:_redirect():2186] Redirects installed.
2025-01-17 16:28:24,900 INFO    MainThread:12039 [wandb_init.py:init():847] run started, returning control to user process
2025-01-17 16:28:31,451 WARNING MsgRouterThr:12039 [router.py:message_loop():77] message_loop has been closed
